### ISSUE TITLE
Improve formatting of generated code

### ISSUE THREAD

**Issue by TiarkRompf** [TiarkRompf/virtualization-lms-core#30]
Original message: The code that is written to output in Delite is not formatted and is very hard to parse. We should a output stream that does the code formatting so it is easier to read. Vlad has a prototype but we should maybe consider some of the existing formatting tools available out there.

We should at least have proper indentation. As a debug option, we should be able to print Defs in internal format as comments. Can we integrate SourceContext info in a useful way?

**Comment by sstucki** [comment]
8f40399add2517d964c61e84cb5ed5da2eedc805 adds basic indentation.

**Comment by TiarkRompf** [comment]
done in 8f40399add2517d964c61e84cb5ed5da2eedc805
